src/preload.js

```diff
@@ -22,6 +28,26 @@ function init() {
   window.MakerBridge = {
     getVersion,
   };
+  window.flashBoardFirmware = flashBoardFirmware;
+}
+
+function flashBoardFirmware(boardType) {
+  if (boardType === BOARD_TYPE.CLASSIC) {
+    let avrgirl = new Avrgirl({board: 'circuit-playground-classic'});
+    window.fetch('https://s3.amazonaws.com/downloads.code.org/maker/CircuitPlaygroundFirmata.ino.circuitplay32u4.hex')
+      .then(function(response) {
+        response.arrayBuffer().then(function(body) {
```

Comment:
Should we do a checksum verification between downloading the hexfile and flashing it to the board.

Reply:
Ahh, good call. I'm having trouble finding resources on how to do this - do you have suggested resources?

Reply:
As a naive pass, I'd probably hard-code an expected md5 hash of the hex file here, and then use something like [blueimp/JavaScript-MD5](https://github.com/blueimp/JavaScript-MD5) or [satazor/js-spark-md5](https://github.com/satazor/js-spark-md5) to check the hash of the response.

Note that using MD5 is no longer considered sufficient to check that a file isn't malicious, due to known collisions.  But it _should_ be good enough to catch a corrupted download, which is the case I think we're more likely to see in the wild.

Further reading:
- https://www.howtogeek.com/363735/what-is-a-checksum-and-why-should-you-care/
- https://security.stackexchange.com/questions/26678/why-is-it-necessary-to-match-the-
checksum-of-a-download-with-another-file-provid

Reply:
> As a naive pass, I'd probably hard-code an expected md5 hash of the hex file here

I'd love for us to be able to update the firmware without needing to do a deploy. Could the checksum of the file go in the filename instead?

Reply:
@joshlory - I'm not familiar with this process. Would it just be having the file named: `<filename>_<checksumValue>.hex` and then we just parse the filename for the checksum value?